src/structures/textChannel.ts

```diff
@@ -185,4 +229,40 @@ export class GuildTextChannel extends TextChannel {
 
     return new GuildTextChannel(this.client, resp, this.guild)
   }
+
+  /**
+   * Bulk Delete Messages in a Guild Text Channel
+   * @param messages Messages to delete. Can be a number, or Array of Message or IDs
+   */
+  async bulkDelete(
+    messages: Array<Message | string> | number
+  ): Promise<GuildTextChannel> {
+    let ids: string[] = []
+
+    if (Array.isArray(messages))
+      ids = messages.map((e) => (typeof e === 'string' ? e : e.id))
+    else {
+      let list = await this.messages.array()
+      if (list.length < messages) list = (await this.fetchMessages()).array()
+      ids = list
+        .sort((b, a) => a.createdAt.getTime() - b.createdAt.getTime())
+        .filter((e, i) => i < messages)
+        .filter(
+          (e) =>
+            new Date().getTime() - e.createdAt.getTime() <=
+            1000 * 60 * 60 * 24 * 14
+        )
+        .map((e) => e.id)
+    }
+
+    ids = [...new Set(ids)]
+    if (ids.length < 2 || ids.length > 100)
```

Comment:
Later we might can make multiple delete requests so it can delete more than 100 messages

Reply:
Wouldn't it be better to let user handle that logic?

Reply:
I don't think so because many beginner developers make that mistake many times

Reply:
Hmm, can we limit it to an additional option then?

Reply:
yeah that's good too